### ISSUE TITLE
[Bug]: Class properties are transpiled even in newer versions of node

### ISSUE THREAD

**Issue by devongovett** [babel/babel#14167]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@babel/cli

### Input code

```js
class Test {
  x = 2;
}
```

### Configuration file name

babel.config.js

### Configuration

```js
module.exports = () => ({
  presets: [
    [
      require('@babel/preset-env'),
      {
        modules: false,
        debug: true,
        targets: {
          node: 14,
        }
      }
    ]
  ]
});
```

### Current and expected behavior

1. Babel transpiles class properties for node versions < 15 even though they are supported in Node 12. [REPL](https://babeljs.io/repl#?browsers=&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=MYGwhgzhAEAqCmEAu0DeAoa0Ae0C80ATANzoC-QA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=Node-14&version=7.16.9&externalPlugins=&assumptions=%7B%7D)

This appears to have started in preset-env v7.14.0. Before that, it was not transpiled. However, I'm confused because looking at the compat data shows:

https://github.com/babel/babel/blob/e7c705a7486cc1a1490d26415336afd3100d3c8f/packages/babel-compat-data/data/plugins.json#L25

### Environment

```
  System:
    OS: macOS 12.0.1
  Binaries:
    Node: 16.6.1 - ~/.nvm/versions/node/v16.6.1/bin/node
    Yarn: 1.22.11 - ~/.nvm/versions/node/v16.6.1/bin/yarn
    npm: 7.20.3 - ~/.nvm/versions/node/v16.6.1/bin/npm
  Monorepos:
    Yarn Workspaces: 1.22.11
    Lerna: 3.22.1
  npmPackages:
    @babel/core: ^7.12.0 => 7.16.7 
    eslint: ^7.20.0 => 7.20.0 
    gulp-babel: ^8.0.0 => 8.0.0 
    lerna: ^3.22.1 => 3.22.1 
```

### Additional context

This caused a performance regression in Parcel because transpiled class private properties run much slower than native.

**Comment by JLHwung** [comment]
The class properties are transpiled by `@babel/plugin-private-methods`, which invokes `@babel/helper-create-class-plugin` to transpile, but not limited to, class private methods. I think we can skip the `@babel/plugin-private-methods` transform when there is no private methods declared in this class. Similar to what I have done in https://github.com/babel/babel/pull/12849.

A more ambitious approach is to gradually transpile class features: https://github.com/babel/babel/pull/13260.

**Comment by devongovett** [comment]
Hmm I had tried it with this and it still transpiled the class properties...

```js
exclude: [
  '@babel/plugin-proposal-private-methods',
  '@babel/plugin-proposal-private-property-in-object',
  '@babel/plugin-proposal-class-static-block'
]
```

**Comment by nicolo-ribaudo** [comment]
Wow this is confusing. If in the repl I set `node 14` in the "targets" fields (rather than setting it directly in the `preset-env` option) it doesn't compile class fields.

**Comment by nicolo-ribaudo** [comment]
This is definitely a bug: `["@babel/preset-env", { "targets": "node 14" }]` doesn't compile class fields, while `["@babel/preset-env", { "targets": { "node": 14 } }]` does.

**Comment by nicolo-ribaudo** [comment]
Ok, this is what's happening:
- Even if they are different plugins, Babel doesn't support transforming private methods when not transforming class fields. For this reason, we "merge" their compatibility data: https://github.com/babel/babel/blob/e7c705a7486cc1a1490d26415336afd3100d3c8f/packages/babel-preset-env/src/plugins-compat-data.ts#L20-L21
- The resulting data marks Node.js 14.6 as the minimum supported version: https://github.com/babel/babel/blob/e7c705a7486cc1a1490d26415336afd3100d3c8f/packages/babel-compat-data/data/plugins.json#L30-L40
- The behavior difference I noticed in https://github.com/babel/babel/issues/14167#issuecomment-1014809535 is because browserslist resolved `node 14` to `14.18`, while we consider the `"node": 14` option as "every Node.js 14 version" (i.e. `>=14.0`).

As a workaround, we can avoid merging the compat data of the two plugins and make `proposal-private-method` automatically also compile class fields in the classes where it's needed (until #13260 is ready).

**Comment by JLHwung** [comment]
> browserslist resolved node 14 to 14.18, while we consider the "node": 14 option as "every Node.js 14 version" (i.e. >=14.0).

Great point! I will add that to the targets docs.
